### PR TITLE
main: add `make server_coverage` build option (#11844)(#12152)

### PR DESCRIPTION
**Pull request by bb7133**
Cherry-pick http://github.com/pingcap/tidb/pull/11844 and http://github.com/pingcap/tidb/pull/12152
to generate `tidb-server-coverage` binary in release 3.0 branch.

The two PRs are trivial and they're closely related, so I decided to cherry-pick them in 1 PR

Conflict list:
```
Makefile
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -13,6 +13,7 @@ export PATH := $(path_to_add):$(PATH)
 
 GO        := GO111MODULE=on go
 GOBUILD   := CGO_ENABLED=1 $(GO) build $(BUILD_FLAG) -trimpath
+GOBUILDCOVERAGE := GOPATH=$(GOPATH) CGO_ENABLED=1 cd tidb-server; $(GO) test -coverpkg="../..." -c .
```
Comment (ngaut) [inline Makefile]: Why enable CGO?

Comment (bb7133) [reply Makefile]: `GOBUILDCOVERAGE` generates a binary that serves the same way with `tidb-server`, this is to make the build as close to `GOBUILD` as possible.

Comment (ngaut) [reply Makefile]: We do not need CGO any more, we should remove it after the PR merged.

Comment (bb7133) [reply Makefile]: Sure~

**Pull request merged**